src/agent/realmain.cpp

```diff
@@ -161,9 +177,19 @@ static void PrintHelp(const char *aProgramName)
             "     --auto-attach          Whether or not to automatically attach to the saved network (default: 1).\n"
             "     --rest-listen-address  Network address to listen on for the REST API (default: 127.0.0.1).\n"
             "     --rest-listen-port     Network port to listen on for the REST API "
-            "(default: " HELP_DEFAULT_REST_PORT_NUMBER ").\n"
-            "\n",
+            "(default: " HELP_DEFAULT_REST_PORT_NUMBER ").\n",
             aProgramName);
+#if OTBR_ENABLE_BORDER_AGENT
+    if (!isVendorNameSet)
```

Comment:
There is no need to check whether the vendor name or the product name is set when printing hel info.